addons/mail/static/src/js/mail_mention.js

```diff
@@ -0,0 +1,224 @@
+ * Open dropdown when user mention partner with @
+ * Dropdown open with highlighted string
+ * Start search if word have minimum length is 3
+ * Wait for at least 400 milliseconds without typing any new letter
+ * Maximum number of partner in dropdown list is 8
+ */
+var Mention = Widget.extend({
+    template: null,
+    init: function(parent) {
+        this._super.apply(this, arguments);
+        this.$textarea = parent.$el.find(".field_text");
+        this.$dropdown = $("<div class='mention'><ul></ul></div>");
+        this.model_res_partner = new Model("res.partner");
+        this.selected_partners = {};
+        this.search_keys = {};
+        this.partners = {};
+        this.keyup_list = [40, 38, 13, 35, 33, 34, 27];
+        this.partner_limit = 8;
+        this.typing_speed = 400;
+        this.min_charactor = 4;
+        this.search_string = '';
+        this.recent_string = '';
+    },
+
+    start: function() {
+        this._super.apply(this, arguments);
+        this.bind_events();
+    },
+    bind_events: function() {
+        var self = this;
+        this.$dropdown.insertAfter(this.$textarea);
+        
+        //textear event binding
+        this.$textarea.on('keyup', this.proxy('textarea_keyup'))
+        .on('click', function(e){ self.clear_dropdown(); })
+        .on('keydown', function(e){
+            var $active = self.$dropdown.find("li.active");
+            if(self.recent_string){
+                if (e.keyCode == 40) {
+                    $active.next().trigger("mouseover");
+                    return false;
+                } else if (e.keyCode == 38) {
+                    $active.prev().trigger("mouseover");
+                    return false;
+                } else if(e.keyCode == 13){
+                    self.find_n_replace(parseInt($active.find("span").attr("id")));
+                    return false;
+                }
+            }
+        });
+        
+        //dropdown event binding
+        this.$dropdown.on('mouseover', 'li', function(e) {
+            $(e.currentTarget).addClass('active').siblings().removeClass();
+        }).on('click', 'li', function(e){
+            self.find_n_replace(parseInt($(e.currentTarget).find("span").attr("id")));
+        });
+        this.clear_dropdown();
+    },
+    show_dropdown: function(){
+        var self = this;
+        var highlight = function(description){
+            if(description)
+            return description.replace(new RegExp(self.search_string, "gi"), function(str) {return _.str.sprintf("<b><u>%s</u></b>",str);});};
+
+        var res = _.map(self.search_keys[self.search_string], function(id) {
+                return {
+                    "id": id, 
+                    "name": highlight(self.partners[id]["name"]), 
+                    "email": highlight(self.partners[id]["email"])
+                    };
+        });
+        this.clear_dropdown();
+        this.$dropdown.find("ul").append(QWeb.render('Mention', {"result": res}));
```

Comment:
You should have 2 templates : one for the list container which will call the second one (the one for the list of searched partners).
Define a list of partner like `this.set('partners', [])`, and add it an event listener `this.on('change:partners, this, this.render_partner)` . When the search rpc will be done, set the result in `partners` attribute. This will trigger `render_partners` method. This method should render the 2nd template and replace the $element in the DOM.